TCMSApp/src/client/app/tests/tests-list.controller.js

```diff
@@ -70,31 +70,34 @@
                 controller: function ($uibModalInstance) {
                     var vmSuite = this;
 
-                    var lastSuite = vm.f[vm.f.length - 1];
-                    vmSuite._id = lastSuite._id + 1;
-                    vmSuite.priority = TestsService.getPriority();
-                    vmSuite.suitePriority = '2';
+                    vmSuite.suiteName = '';
+                    vmSuite.suiteDescription = '';
                     vmSuite.cancelAddSuite = cancelAddSuite;
                     vmSuite.submitAddSuite = submitAddSuite;
 
                     function submitAddSuite() {
-                        var suite = {};
+                        var Suite = TestsService.getSuites(),
+                            suite = new Suite();
                         suite.suiteName = vmSuite.suiteName;
-                        suite._id = vmSuite._id;
                         suite.suiteDescription = vmSuite.suiteDescription;
-                        suite.suitePriority = vmSuite.suitePriority;
-                        suite.tests = [];
-                        vm.f.push(suite);
-                        vm.currentSuite = suite;
-                        TestsService.setCurrentSuite(vm.currentSuite);
-                        logger.success('New Suite created');
-                        $uibModalInstance.dismiss();
+                        suite.project = user.currentProjectID;
+                        var resource = TestsService.checkForSameSuiteName(suite.suiteName, suite.project);
+                        resource.$promise.then(function (res) {
+                            if (res.length) {
+                                logger.error('Test suite with the same name alrady exists');
+                            }
+                            else {
+                                suite.$save();
```

Comment:
we have a bit different style of the checking for the same name. It can be checked as a validation on save. See Alexandra's solution. It's cleaner.
here https://github.com/ITsvetkoFF/Kv-012/blob/Defects_workflow2/TCMSApp/src/client/app/defects/add-defect.controller.js from line 36